File: programmers/lv2_review/lv2_4.py
# 주식가격
# 스택 큐를 사용하여 문제 해결하는 방법?
def solution(prices):
    res = [0] * len(prices)
    for i in range(len(prices)):
        idx = len(prices) - 1
        for j in range(i, len(prices)):
            if prices[i] > prices[j]:
                idx = j
                break
        res[i] = idx - i
    return res


a = [1, 2, 3, 2, 3]
print(solution(a))

File: programmers/lv2_review/lv2_4.js
function solution(n) {
  if (n === 0) return "";
  let string = "";
  if (n % 3 === 1) {
    string = "1";
  } else if (n % 3 === 2) {
    string = "2";
  } else if (n % 3 === 0) {
    string = "4";
  }
  if (n % 3 === 0) {
    string = solution(Math.floor(n / 3) - 1) + string;
  } else {
    string = solution(Math.floor(n / 3)) + string;
  }
  return string;
}

console.log(solution(4));
